File: src/main/scala/com/github/mauricio/postgresql/column/TimestampWithTimezoneEncoderDecoder.scala
package com.github.mauricio.postgresql.column

import org.joda.time.format.DateTimeFormat

/**
 * User: Maurício Linhares
 * Date: 3/6/12
 * Time: 9:27 AM
 */

object TimestampWithTimezoneEncoderDecoder extends TimestampEncoderDecoder {

  private val format = DateTimeFormat.forPattern("yyyy-MM-dd HH:mm:ss.SSSSSSZ")

  override def formatter = format

}

File: db-async-common/src/main/java/com/github/mauricio/async/db/column/TimestampWithTimezoneEncoderDecoder.kt

package com.github.mauricio.async.db.column

import org.joda.time.DateTime
import org.joda.time.format.DateTimeFormat

object TimestampWithTimezoneEncoderDecoder : TimestampEncoderDecoder() {

  private val format = DateTimeFormat.forPattern("yyyy-MM-dd HH:mm:ss.SSSSSSZ")

  override fun formatter ()= format

  override fun decode(value: String): Any {
    return formatter().parseDateTime(value)
  }

}
